Initialize scripts

jackboberg opened the pull request:
Use build-base and bootstrap script to install dependencies. Build
script sources dnvm, update coreclr, and restores app dependencies. Home
directory is copied into build output for use at runtime.

jackboberg committed:
Initialize scripts

Use build-base and bootstrap script to install dependencies. Build
script sources dnvm, update coreclr, and restores app dependencies. Home
directory is copied into build output for use at runtime.

jackboberg commented:
I would like to replace all instances of `cp -R ...` in build script with `mv` calls to make it faster.

kenXengineering commented on bin/build:
```diff
@@ -0,0 +1,35 @@
+fi
+
+mkdir -p $INPUT_DIR
+mkdir -p $OUTPUT_DIR
```
Don't create `$OUTPUT_DIR` as we will be creating it later when we move input to output.

jackboberg replied:
The output directory is mounted in before `build` is called, this line exists to make some elements of testing locally easier.

kenXengineering commented on bin/build:
```diff
@@ -0,0 +1,35 @@
+mkdir -p $INPUT_DIR
+mkdir -p $OUTPUT_DIR
+
+if [[ ! -d $INPUT_DIR ]] || [[ ! -d $OUTPUT_DIR ]]; then
```
Don't check for `$OUTPUT_DIR` as it will not exist.

kenXengineering commented on bin/build:
```diff
@@ -0,0 +1,35 @@
+cp -R /opt/home/. $HOME
+
+# mv $INPUT_DIR/{*,.*} $OUTPUT_DIR > /dev/null 2>&1
+cp -R $INPUT_DIR/. $OUTPUT_DIR
```
This can be `mv $INPUT_DIR $OUTPUT_DIR` since both _should_ be in the `/mnt` dir.  If they aren't then it acts like a copy.

kenXengineering commented on bin/build:
```diff
@@ -0,0 +1,35 @@
+echo "dnu restore"
+dnu restore
+
+cp -R $HOME/. $MOD_DIR/home
```
This can be `mv $HOME $MOD_DIR/home` since both _should_ be in the `/mnt` dir.  If they aren't it acts like a copy.  `$MOD_DIR/home` shouldn't exist before moving it.

kenXengineering commented on bootstrap.sh:
```diff
@@ -0,0 +1,31 @@
+cd ~/ && ldconfig
+
+# clean up
+apt-get autoclean && apt-get autoremove -y && apt-get clean
```
You can add `rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*` to this as well to clear out tmp files and folders.

jackboberg replied:
added

jackboberg committed:
Move $HOME instead of copying it

jackboberg committed:
Clear out tmp files and folders

jackboberg committed:
Remove output dir and move input before building

This will speed up build times, since we don't have to copy the
directory. It will also decrease volume size during a build.

Reorganized some of the script to group commands.

jackboberg committed:
Consume verbose output of upgrade

This command attempts to draw a progress bar, which does not work well
with streamed logging.
